### PR TITLE
Adds a prework file on the backend training for the Dojo

### PR DESCRIPTION
**Pull request by kurenn**
### What does this PR do?

* Adds the prework necessary for the backend training on Ruby

### CODE REVIEW

File: backend-training/PREWORK.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    <li>What is an <a target="_blank" href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29">abstract class?</a></li>
+    <li>Difference between a <a target="_blank" href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29#The_concept_of_class_interface">Interface</a> and an <a href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29">abstract class?</a></li>
+    <li>What is a <a target="_blank" href="http://en.wikipedia.org/wiki/Class_diagram">Class Diagram?</a></li>
+    <li>Check this <a target="_blank" href="http://en.wikipedia.org/wiki/Data_structure">data structures</a></li>
+  </ul>
+</ul>
+
+### Software requirements
+
+We provide a list of necessary software you need to install before starting your training.
```
Comment (miguejs) [inline backend-training/PREWORK.md]: We need to add a note that says "if your current OS is windows you can setup a linux vm  or install linux in another partition,  windows is not allowed in our development workflow  at the moment

Comment (kurenn) [reply backend-training/PREWORK.md]: Good one! Let me add it!

File: backend-training/PREWORK.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <a target="_blank" href="http://en.wikipedia.org/wiki/Object_composition#Composite_types_in_C">composition?</a></li>
+    <li>What is <a target="_blank" href="http://en.wikipedia.org/wiki/Inheritance_%28object-oriented_programming%29">inheritance</a>?</li>
+    <li>What is an <a target="_blank" href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29">abstract class?</a></li>
+    <li>Difference between a <a target="_blank" href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29#The_concept_of_class_interface">Interface</a> and an <a href="http://en.wikipedia.org/wiki/Class_%28computer_programming%29">abstract class?</a></li>
+    <li>What is a <a target="_blank" href="http://en.wikipedia.org/wiki/Class_diagram">Class Diagram?</a></li>
+    <li>Check this <a target="_blank" href="http://en.wikipedia.org/wiki/Data_structure">data structures</a></li>
+  </ul>
+</ul>
+
+### Software requirements
```
Comment (miguejs) [inline backend-training/PREWORK.md]: Do we need to add docker as a software requirement  too ?

Comment (kurenn) [reply backend-training/PREWORK.md]: Probably not for this one, I would suggest to add another set of training tools or dojo course to meet this concern, what do you think?

File: backend-training/PREWORK.md (hunk 1 above)
Comment (kurenn) [reply backend-training/PREWORK.md]: Done!

File: backend-training/PREWORK.md (hunk 2 above)
Comment (miguejs) [reply backend-training/PREWORK.md]: I think you are right, docker needs to have his own course or something like that

**Pull request merged**